### PR TITLE
Draft of Enterprise Dashboard

### PR DESCRIPTION
**Pull request by orangejenny**
[Spec](https://docs.google.com/document/d/1kBxM5guH8vxNByS06BtRaqgGQkzeKULbp0MMwF01xfo/edit)

This is currently superuser-only and is linked from a new tab on the account info page, which is all temporary. Planning to update that, and to hook up the menu, once the permissioning / Custom Account work is done.

Review by commit.

@nickpell / @gbova 
code buddy @gcapalbo 

screen shot 2018-05-30 at 6 00 17 pm

### CODE REVIEW

File: corehq/apps/accounting/enterprise.py [hunk 1]
```diff
@@ -63,30 +76,85 @@ def rows_for_domain(self, domain):
...
+                                     lt=datespan.enddate_adjusted))
+                 .filter(users_filter))
+        rows = []
+        for hit in query.run().hits:
+            username = hit['form']['meta']['username']
+            submitted = self.format_date(datetime.strptime(hit['received_on'][:19], '%Y-%m-%dT%H:%M:%S'))
+            rows.append([
+                hit['form']['@name'],
+                submitted,
+                apps[hit['app_id']] if hit['app_id'] in apps else 'App not found',
```
Comment (gcapalbo) [inline corehq/apps/accounting/enterprise.py]: should `App not found` be translated?

File: corehq/apps/accounting/enterprise.py [hunk 2]
```diff
@@ -63,30 +76,85 @@ def rows_for_domain(self, domain):
...
     @property
     def headers(self):
         headers = super(EnterpriseWebUserReport, self).headers
         return ['Name', 'Email Address', 'Role', 'Last Login'] + headers
 
+    def rows_for_domain(self, domain):
+        rows = []
+        for user in get_all_user_rows(domain.name, include_web_users=True, include_mobile_users=False,
+                                      include_inactive=False, include_docs=True):
+            user = WebUser.wrap(user['doc'])
```
Comment (gcapalbo) [inline corehq/apps/accounting/enterprise.py]: I've recently gotten myself into the habit of always using `CouchUser.wrap_correctly`, the reason being that the user classes unfortunately have lazy migrations with side effects on wrap, which are different for both user class types. So on the off chance that this result is mismatched some unexpected behavior might happen during the lazy migration, and using wrap_correctly prevents that.

File: corehq/apps/accounting/enterprise.py [hunk 3]
```diff
@@ -63,30 +76,85 @@ def rows_for_domain(self, domain):
...
     @property
     def headers(self):
         headers = super(EnterpriseMobileWorkerReport, self).headers
         return ['Username', 'Name', 'Last Sync', 'Last Submission', 'CommCare Version'] + headers
 
+    def rows_for_domain(self, domain):
+        rows = []
+        for user in get_all_user_rows(domain.name, include_web_users=False, include_mobile_users=True,
+                                      include_inactive=False, include_docs=True):
+            user = CommCareUser.wrap(user['doc'])
```
Comment (gcapalbo) [inline corehq/apps/accounting/enterprise.py]: (same comment about wrap_correctly)

**Comment by gcapalbo** [comment]
@orangejenny you may be thinking of this already for a later pr, but was just curious how long it takes to produce the report and its downloaded version, and if you think those should be made asynchronous

File: corehq/apps/accounting/enterprise.py (hunk 2 above)
Comment (orangejenny) [reply corehq/apps/accounting/enterprise.py]: I didn't know about `wrap_correctly`, thanks

**Pull request merged**